src/app/api/chapter-request/route.spec.ts

```diff
@@ -0,0 +1,47 @@
+import { describe, expect, test } from "vitest";
+import { ChapterRequestResponse, ChapterRequest } from "./route.schema";
+import { chapterRequest } from "./route.client";
+
+describe("ChapterRequest", () => {
+  test("InvalidEmailThrowsError", () => {
+    expect(() =>
+      ChapterRequest.parse({
+        firstName: "John",
+        lastName: "Doe",
+        universityEmail: "not an email",
+        phoneNumber: "1234567890",
+        university: "University of Waterloo",
+        universityAddress: "123 University Ave",
+        leadershipExperience: "I have lots of leadership experience",
+        motivation: "I am very motivated",
+        availabilities: "I am available",
+        questions: "I have a question",
+      })
+    ).toThrowError("This is not a valid email");
+  });
+
+  test("DuplicateEmailThrowsError", async () => {
+    const body = {
+      firstName: "Don't delete me",
+      lastName: "Don't delete me",
+      universityEmail: "donaldduck@gmail.com",
+      phoneNumber: "1234567890",
+      university: "University of Waterloo",
+      universityAddress: "123 University Ave",
+      leadershipExperience: "I have lots of leadership experience",
+      motivation: "I am very motivated",
+      availabilities: "I am available",
+      questions: "I have a question",
+    };
+    const response = await fetch("http://localhost:3000/api/chapter-request", {
```

Comment:
Does this test run?

Reply:
Yep

Reply:
Ah you have to start a server at the same time too right? I think we should avoid network calls in unit tests. An alternative option is exploring how to mock Prisma client to replicate "duplicate email". What do you think?